src/Storages/ObjectStorageQueue/ObjectStorageQueueMetadataFactory.cpp

```diff
@@ -30,42 +31,67 @@ ObjectStorageQueueMetadataFactory::FilesMetadataPtr ObjectStorageQueueMetadataFa
         auto & metadata_from_keeper = it->second.metadata->getTableMetadata();
 
         metadata_from_table.checkEquals(metadata_from_keeper);
-
-        it->second.ref_count += 1;
     }
+
+    it->second.metadata->registerIfNot(storage_id);
```

Comment:
no need for try catch block here ?

Reply:
No, it is ok to fail table creation in this case, we anyway to other stuff in keeper in constructor, not only this, without them we cannot start.